grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractUnframedGrpcService.java

```diff
@@ -213,11 +215,7 @@ static void deframeAndRespond(ServiceRequestContext ctx,
         final ResponseHeadersBuilder unframedHeaders = grpcResponse.headers().toBuilder();
         unframedHeaders.set(GrpcHeaderNames.GRPC_STATUS, grpcStatusCode); // grpcStatusCode is 0 which is OK.
         if (grpcMediaType != null) {
-            if (grpcMediaType.is(GrpcSerializationFormats.PROTO.mediaType())) {
-                unframedHeaders.contentType(MediaType.PROTOBUF);
-            } else if (grpcMediaType.is(GrpcSerializationFormats.JSON.mediaType())) {
-                unframedHeaders.contentType(MediaType.JSON_UTF_8);
-            }
+            unframedHeaders.contentType(originalContentType);
```

Comment:
for requests that use x-protobuf or x-google-protobuf, my guess is that they would expect the server to respond with the same content type. Would using the passed down originalContentType be safe? Also, should accept header be checked for the response protobuf content type?

Reply:
> Would using the passed down originalContentType be safe? Also, should accept header be checked for the response protobuf content type?

If a suitable `accept-header` is specified, it makes sense to respect the value over `content-type`.

> Also, should there be an else clause here? (for when grpcMediaType is null)

It looks like defensive code.
A normal response including an `OK` status should be returned with a proper `content-type`.
https://github.com/line/armeria/blob/49054ad7f22a7f2d64faaeb694674b0df09ad165/grpc/src/main/java/com/linecorp/armeria/server/grpc/FramedGrpcService.java#L181-L187